test/disk_uploader_test.go

```diff
@@ -0,0 +1,150 @@
+package test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/utils/ptr"
+
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	kubevirtcliv1 "kubevirt.io/client-go/kubecli"
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/constants"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/framework"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/runner"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/testconfigs"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/vm"
+)
+
+var _ = Describe("Run disk-uploader", func() {
+	f := framework.NewFramework()
+
+	BeforeEach(func() {
+		if f.TestOptions.SkipDiskUploaderTests {
+			Skip("skipDiskUploaderTests is set to true, skipping tests")
+		}
+	})
+
+	It("Extracts disk from VM and upload to container registry", func() {
+		vmName := constants.E2ETestsRandomName("example-vm")
+		dvName := constants.E2ETestsRandomName("example-dv")
+		secretName := constants.E2ETestsRandomName("disk-uploader-credentials")
+
+		config := &testconfigs.DiskUploaderTestConfig{
+			TaskRunTestConfig: testconfigs.TaskRunTestConfig{},
+			TaskData: testconfigs.DiskUploaderTaskData{
+				ExportSourceKind: "vm",
+				ExportSourceName: vmName,
+				VolumeName:       dvName,
+				ImageDestination: "quay.io/boukhano/e2e-example-vm", // TODO: Use quay.io/kubevirt/e2e-example-vm
```

Comment:
Is there a registry local to the test cluster, that can be used?

Reply:
I had a discussion with @ksimon1 about it to keep using an external registry that allows you also to test it on Kubernetes cluster. As you know, only OpenShift has built-in container registry.